mex/common/types/temporal_entity.py

```diff
@@ -0,0 +1,419 @@
+    ) -> None:
+        """Validate for precision (year or month) after initialization."""
+        super().__init__(*args, tzinfo=tzinfo)  # type: ignore
+        if self.precision not in (
+            TemporalEntityPrecision.YEAR,
+            TemporalEntityPrecision.MONTH,
+        ):
+            raise ValueError("Expected precision level 'YEAR' or 'MONTH'")
+
+    @classmethod
+    def __get_pydantic_core_schema__(
+        cls, _source: Type[Any], _handler: GetCoreSchemaHandler
+    ) -> core_schema.CoreSchema:
+        """Mutate the field schema for year-month pattern."""
+        from_str_schema = core_schema.chain_schema(
+            [
+                core_schema.str_schema(pattern=YEAR_MONTH_REGEX),
+                core_schema.no_info_plain_validator_function(
+                    cls.validate,
+                ),
+            ]
+        )
+
+        from_anything_schema = core_schema.chain_schema(
+            [
+                core_schema.no_info_plain_validator_function(cls.validate),
+                core_schema.is_instance_schema(YearMonth),
+            ]
+        )
+
+        return core_schema.json_or_python_schema(
+            json_schema=from_str_schema,
+            python_schema=from_anything_schema,
+        )
+
+    @classmethod
+    def __get_pydantic_json_schema__(
+        cls, core_schema_: CoreSchema, handler: GetJsonSchemaHandler
+    ) -> JsonSchemaValue:
+        """Modify the schema to add the class name as title and examples."""
+        json_schema = super().__get_pydantic_json_schema__(core_schema_, handler)
+        json_schema["examples"] = ["2011", "2019-03"]
+        return json_schema
+
+
+class YearMonthDay(TemporalEntity):
+    """Date pattern that accepts only yyyy-MM-dd format."""
+
+    def __init__(
+        self,
+        *args: Union[int, str, date, datetime, "YearMonthDay"],
+        tzinfo: Optional[tzinfo] = None,
+    ) -> None:
+        """Validate for day precision after initialization."""
+        super().__init__(*args, tzinfo=tzinfo)  # type: ignore
+        if self.precision != TemporalEntityPrecision.DAY:
+            raise ValueError("Expected precision level 'DAY'")
+
+    @classmethod
+    def __get_pydantic_json_schema__(
+        cls, core_schema_: CoreSchema, handler: GetJsonSchemaHandler
+    ) -> JsonSchemaValue:
+        """Modify the schema to add the class name as title and examples."""
+        json_schema = super().__get_pydantic_json_schema__(core_schema_, handler)
+        json_schema["examples"] = ["2014-08-24"]
+        json_schema["format"] = "date"
+        return json_schema
+
+    @classmethod
+    def __get_pydantic_core_schema__(
+        cls, _source: Type[Any], _handler: GetCoreSchemaHandler
+    ) -> core_schema.CoreSchema:
+        """Mutate the field schema for date pattern."""
+        from_str_schema = core_schema.chain_schema(
+            [
+                core_schema.str_schema(pattern=YEAR_MONTH_DAY_REGEX),
```

Comment:
maybe put this as a class var `STR_SCHEMA_PATTERN = YEAR_MONTH_DAY_REGEX` and then we can have just 1 `__get_pydantic_core_schema__` instead of 4

Reply:
Great idea, done in: https://github.com/robert-koch-institut/mex-common/commit/57c0b499a69f46a0a0a7fdc75a7a5b5c8fd94c63